fibonacci.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Template Name: fibonacci
+ * Template Post Type: post, page
+ *
+ * @package WordPress
+ * @subpackage fibonacci
+ * @since fibonacci
+ */
+function mathFibonacci($n)
+{
+    return round(pow((sqrt(5) + 1) / 2, $n) / sqrt(5));
+}
+
+function loopFibonacci($n)
+{
+    $first = 0;
+    $second = 1;
+
+    for ($i = 1; $i < $n; $i++) {
+
+        $sum = $first + $second;
+        $first = $second;
+        $second = $sum;
+    }
+    return $sum;
+}
+
+function tailRecursionFibonacci($n)
+{
+    if ($n < 3) {
+        return 1;
+    } else {
+        return TailRecursionFibonacci($n - 1) + TailRecursionFibonacci($n - 2);
```

Comment:
This is not tail recursion. The last operation of tail recursion should be returning. Your last operation is sum.